lib/rspec-puppet-facts.rb

```diff
@@ -228,7 +229,13 @@ def self.with_custom_facts(os, facts)
       next if fact[:options][:confine] && !fact[:options][:confine].include?(os)
       next if fact[:options][:exclude] && fact[:options][:exclude].include?(os)
 
-      facts[name] = fact[:value].respond_to?(:call) ? fact[:value].call(os, facts) : fact[:value]
+      value = fact[:value].respond_to?(:call) ? fact[:value].call(os, facts) : fact[:value]
+      # if merge_facts passed, merge supplied facts into facts hash
+      if fact[:options][:merge_facts]
+        facts = facts.deep_merge!({name => value})
```

Comment:
If you already use `deep_merge!` you're changing it in place, so this is sufficient:
```suggestion
        facts.deep_merge!({name => value})
```

Reply:
Updated.